plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -255,6 +271,32 @@ private Object toIntegerVarcharMapBlock(Map<Integer, String> values)
             blockBuilder.closeEntry();
             return integerToVarcharMapType.getObject(blockBuilder, 0);
         }
+
+        private Object getBigintArrayBlock(List<Long> values)
+        {
+            if (values == null) {
+                return null;
+            }
+
+            BlockBuilder blockBuilder = bigintArrayType.createBlockBuilder(null, values.size());
+            BlockBuilder singleArrayBlockBuilder = blockBuilder.beginBlockEntry();
+            values.forEach(value -> BIGINT.writeLong(singleArrayBlockBuilder, value));
+            blockBuilder.closeEntry();
+            return bigintArrayType.getObject(blockBuilder, 0);
+        }
+
+        private Object getIntegerArrayBlock(List<Integer> values)
+        {
+            if (values == null) {
+                return null;
+            }
+
+            BlockBuilder blockBuilder = integerArrayType.createBlockBuilder(null, values.size());
```

Comment:
expected entries are 1 block of List type.

```suggestion
            BlockBuilder blockBuilder = integerArrayType.createBlockBuilder(null, 1);
```